python/paddle/fluid/framework.py

```diff
@@ -3937,6 +3937,10 @@ def __init__(self):
         # appending gradients times
         self._appending_grad_times = 0
 
+        # identifier for auto checkpoint
+        self._auto_checkpoint = True
+        self._auto_checkpoint_name = None
```

Comment:
why these two attributes spread in `compiler.py`, `executor.py` and `framework.py`

Reply:
is it better to check whether the program has op of backword roles to determine a train program? it will make less effect to current code.

Reply:
1. Every program should have a unique name to identify it.
2. Check the program's content will waste time?

Reply:
Cleanup them. Done.